website/content/docs/operations/stateful-workloads.mdx

```diff
@@ -0,0 +1,116 @@
+
+## Considerations
+There are a few considerations when choosing the most appropriate storage strategy, such as performance, reliability, availability, and maintenance.
+
+Local storage is usually the most performant and available option, requiring little maintenance outside of ensuring adequate capacity. However, since there is no redundancy, if a single node, disk, or group of disks fails, data loss and service interruption will occur. A geographically distributed networked storage with multiple redundancies, including disks, controllers, and network paths, would be the most highly available and reliable. 
+
+For high availability and resilience, use a geographically distributed networked storage with multiple redundancies, including disks, controllers, and network paths. This implementation can tolerate multiple hardware failures before risking data loss. However, it relies on network performance and comes with the tradeoffs of higher latency, lower throughput, and increased maintenance. 
+
+This document addresses traditional storage you can attach as a filesystem and does not cover object storage such as AWS S3 or MinIO. 
+
+It is also important to consider the environment Nomad will run in when reviewing different storage options. Nomad can run either in public clouds or on-prem. Depending on your organizational and application needs, understanding the storage options for each provider will help you optimize your stateful workloads.
+
+
+### Public cloud
+
+Public cloud providers offer services, such as Azure Ephemeral OS disks, Managed Disks and Files, GCP Local SSDs, Persistent Disk, and Filestore, and AWS EBS. The relevant difference is that Azure Managed Disks and GCP Persistent Disks are optionally regional, unlike AWS EBS.
+
+In an AWS environment, an EBS block volume is zonal and slower than the local storage on the EC2 instances, but it is readily available. Your data will persist in the case of a crash of the underlying hardware, and you can reattach the volume to a different EC2 instance within the same AZ. However, in the case of a zonal failure, the data on it is "stuck" within that zone, and the application is not redundant across Availability Zones. You must provision volumes at a fixed size, which you need to monitor and manage. 
+
+AWS EFS is a regional service that provides file storage using the NFS protocol. Since the data replicates across different Availability Zones and uses a higher level protocol (file instead of block like EBS), EFS has higher latencies and lower throughput than EBS. However, it can continue to serve data requests in case of a zonal failure. A significant advantage, and sometimes a disadvantage, for NFS and similar protocols is that you can mount and access them simultaneously on multiple machines.
```

Comment:
Do we really want to talk about a specific cloud vendor's services here, as opposed to the general type of volumes with these as an example? Ex. should we talk about the difference between local block vs network filesystems, rather than EBS vs EFS?

Reply:
@tgross my thinking here was to give concrete examples with the cloud providers' services names, with the respective caveats (e.g. EBS being zonal thus being unsuitable for a lot of  persistent storage needs), to make it clear which cloud service correspond to which storage option. However I see the whole paragraph has gotten a little mixed and I'll reword it to make clear which services are on which cloud provider and what they do.

Reply:
I tried doing that, and after some back and forth decided to go with a table format for better readability. What do you think about the following for AWS (wanted some feedback before doing it for GCP and Azure too):
```
### AWS 

| AWS service                                                                                  | Availability                                                      | Persistence                                                                      | Performance                                                                                                                                        | Suitability                                                                                                                  | Use in Nomad                               |
|----------------------------------------------------------------------------------------------|-------------------------------------------------------------------|----------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------|
| [Instance Storage](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html) | Locally on some instance types                                    | Limited, not persistent accross instance stops/terminations or hardware failures | High throughput and low latency                                                                                                                    | Temporary storage of information that changes frequently, such as buffers, caches, scratch data, and other temporary content | via `ephemeral_disk` or `host_volume`      |
| [Elastic Block Store](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AmazonEBS.html)    | Zonal block devices attached to instances                         | Strongly persistent, with an independent lifecycle                               | [Configurable](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSPerformance.html), but higher latency than Instance Store                    | General purpose persistent storage                                                                                           | via the CSI plugin or static `host_volume` |
| [Elastic File System](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AmazonEFS.html)    | Regional file storage that can be available to multiple instances | Strongly persistent, with an independent lifecycle                               | [Configurable](https://docs.aws.amazon.com/efs/latest/ug/performance.html), but with less throughput and higher latency than Instance Store or EBS | File storage that needs to be available to multiple instances in multiple zones (even only as a failover)                    | via the CSI plugin or static `host_volume` |


```